src/Agent.Listener/CommandSettings.cs

```diff
@@ -251,13 +254,19 @@ public string GetToken()
                 validator: Validators.NonEmptyValidator);
         }
 
-        // TODO: REMOVE defaultValue parameter after fix in master and rebase on master
-        public string GetUrl(string defaultValue = null)
+        public string GetUrl(bool suppressPrompt = false)
```

Comment:
Rename `suppressPrompt` to `requireValue`. When requireValue=false, if GetArg returns a value, then call GetArgOrPrompt so the validator is run. Otherwise return.